blockadstv/src/main/java/app/pwhs/blockadstv/service/TvVpnService.kt

```
@@ -0,0 +1,304 @@
+                    protocol = dnsProtocol.name,
+                    primary = upstreamDns,
+                    fallback = fallbackDns,
+                    dohUrl = dohUrl,
+                )
+                goTunnelAdapter.setBlockResponseType(dnsResponseType)
+
+                vpnInterface?.let {
+                    goTunnelAdapter.start(
+                        vpnInterface = it,
+                        socketProtector = { fd ->
+                            try { protect(fd) } catch (e: Exception) { false }
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)